### PR TITLE
Add libnuma/2.0.14

### LINKED ISSUE #508: [request] libnuma/2.0.13
### Package Details
  * Package Name/Version: **numa/2.0.13**
  * Source code: **https://github.com/numactl/numactl**

### Description Of The Library / Tool

The libnuma library offers a simple programming interface to the NUMA (Non Uniform Memory Access) policy supported by the Linux kernel.

- This package is an optional requirement for Facebook Rocksdb (#410)
- It contains an executable, but I think it can be excluded from the package. Out proposal is use the library only.

### PR DESCRIPTION
**Pull request by theirix**
Specify library name and version:  **libnuma/2.0.14**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

Closes #508

### CODE REVIEW

File: recipes/libnuma/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import os
+
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+
+class LibnumaConan(ConanFile):
+    name = "libnuma"
+    description = "NUMA support for Linux."
+    license = "GPL-2.0-or-later"
```
Comment (prince-chrismc) [inline recipes/libnuma/all/conanfile.py:9]: https://github.com/numactl/numactl#license-copyrights-acknowledgements

Looking this over I see there is only the `libnuma` that's packaged which is LGPL...

Does this make sense

```suggestion
    license = "LGPL-2.1-or-later"
```

File: recipes/libnuma/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration("{} is only supported on Linux".format(self.name))
```
Comment (prince-chrismc) [inline recipes/libnuma/all/conanfile.py:36]:
```suggestion
        if not self.settings.os in ["Linux", "FreeBSD"]:
            raise ConanInvalidConfiguration("{} is only supported on Linux/FreeBSD".format(self.name))
```

File: recipes/libnuma/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        self._autotools.configure(args=args, configure_dir=self._source_subfolder)
+        return self._autotools
+
+    def build(self):
+        self._patch_sources()
+        autotools = self._configure_autotools()
+        autotools.make()
+
+    def package(self):
+        self.copy("LICENSE.GPL2", dst="licenses", src=self._source_subfolder)
```
Comment (prince-chrismc) [inline recipes/libnuma/all/conanfile.py:73]: as per above

```suggestion
        self.copy("LICENSE.LGPL2.1", dst="licenses", src=self._source_subfolder)
```

**Review by prince-chrismc — COMMENTED** [review]
Tiny comments! Please confirm the lib packaged in regards to the license

File: recipes/libnuma/all/conanfile.py (hunk 1 above)
Comment (theirix) [reply recipes/libnuma/all/conanfile.py:9]: There are two licenses in the repository. I missed the fact in README that libnuma is a subject for less strict LGPL.  Thanks for pointing out!

File: recipes/libnuma/all/conanfile.py (hunk 2 above)
Comment (theirix) [reply recipes/libnuma/all/conanfile.py]: libnuma is not supported on FreeBSD. maybe NUMA as technology is available but `shm.c` fails to compile due to missing `fstat64`, `mmap64`. Reverting.

**Comment by theirix** [comment]
Hmm, hook 'FPIC MANAGEMENT (KB-H007)' complaints about Windows even when I raise `ConanInvalidConfiguration` for non-Linux.

**Comment by theirix** [comment]
Something wrong with build attempts on Windows with a `validate` disabling them

**Comment by prince-chrismc** [comment]
😠 Rouge windows agents!

Very weird.

**Review by prince-chrismc — DISMISSED** [review]

**Comment by prince-chrismc** [comment]
Maybe switch back to configure

https://github.com/conan-io/conan-center-index/pull/4254/files#r557516012

**Comment by prince-chrismc** [comment]
https://github.com/conan-io/conan-center-index/pull/4133#discussion_r557587265

**Review by prince-chrismc — DISMISSED** [review]

File: recipes/libnuma/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        tools.rmdir(os.path.join(self.package_folder, "bin"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "share"))
+        tools.remove_files_by_mask(os.path.join(self.package_folder, "lib"), "*.la")
+
+    def package_info(self):
+        self.cpp_info.components["numa"].libs = ["numa"]
+        self.cpp_info.components["numa"].names["pkg_config"] = "numa"
+        self.cpp_info.components["numa"].names["pkg_config"] = "numa"
+        self.cpp_info.components["numa"].system_libs.extend(["dl", "pthread"])
```
Comment (SpaceIm) [inline recipes/libnuma/all/conanfile.py:85]:
```suggestion
        self.cpp_info.libs = ["numa"]
        self.cpp_info.names["pkg_config"] = "numa"
        self.cpp_info.system_libs = ["dl", "pthread"]
```

File: recipes/libnuma/all/conanfile.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    _autotools = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
```
Comment (SpaceIm) [inline recipes/libnuma/all/conanfile.py:35]:
```suggestion
```

dead code

File: recipes/libnuma/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
+import os
+
+from conans import ConanFile, AutoToolsBuildEnvironment, tools
+from conans.errors import ConanInvalidConfiguration
+
+required_conan_version = ">=1.32.0"
```
Comment (SpaceIm) [inline recipes/libnuma/all/conanfile.py:6]:
```suggestion
required_conan_version = ">=1.29.1"
```

for `tools.remove_files_by_mask()`

File: recipes/libnuma/all/conanfile.py (hunk 4 above)
Comment (theirix) [reply recipes/libnuma/all/conanfile.py:81]: Yes, I meant cmake in L84.

What is the general approach to using or omitting components in recipes? I see the following cases:
- at least two libs
- library is typically used in other projects as a cmake target

Comment (SpaceIm) [reply recipes/libnuma/all/conanfile.py:81]: But this lib doesn't export CMake config files, and there is only one library. conan components are useless here.

Comment (theirix) [reply recipes/libnuma/all/conanfile.py:81]: Exactly. Just clarifying for future submissions.

**Comment by theirix** [comment]
Another strange thing - CI check has failed: "license/cla Expected — Waiting for status to be reported"

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**